### PR TITLE
fix: SqlEditor_spec tests

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Fixes SqlEditor_spec tests:
- Waits for elements to be rendered
- Removes fetch warnings with `fetch-mock`
- Provides required prop `addDangerToast`

### TEST PLAN
1 - Execute tests in spec/javascripts/sqllab/SqlEditor_spec
2 - All tests should pass

@rusackas 

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
@eschutho might want to take a gander at this one too :)

File: superset-frontend/spec/javascripts/sqllab/SqlEditor_spec.jsx [hunk 1]
```diff
@@ -72,22 +78,26 @@ describe('SqlEditor', () => {
       },
     );
 
-  it('render a SqlEditorLeftBar', () => {
+  it('render a SqlEditorLeftBar', async () => {
```
Comment (rusackas) [inline superset-frontend/spec/javascripts/sqllab/SqlEditor_spec.jsx:81]: I'm not sure some tests in this file were skipped in the first place, but I wonder if they work now :P

**Review by rusackas — APPROVED** [review]
PR LGTM. We can circle back to auditing/fixing/removing skipped tests in another bash, but I am a little curious if this change stabilizes them by some miracle.

File: superset-frontend/spec/javascripts/sqllab/SqlEditor_spec.jsx (hunk 1 above)
Comment (michael-s-molina) [reply superset-frontend/spec/javascripts/sqllab/SqlEditor_spec.jsx:81]: They don't work. I think they are really old tests. One of them is using setState but the component is a functional component now and it seems that the height logic being tested is also different.

**Pull request merged**